zerver/management/commands/sync_ldap_user_data.py

```diff
@@ -77,4 +78,10 @@ def handle(self, *args: Any, **options: Any) -> None:
             user_profiles = self.get_users(options, realm, is_bot=False, include_deactivated=True)
         else:
             user_profiles = UserProfile.objects.select_related().filter(is_bot=False)
+
+        if not user_profiles:
+            raise CommandError(
+                "No relevant Zulip users found for syncing. Users need to first be created."
```

Comment:
Not sure if that's the best error message. This was proposed:

```quote
I think a check like that might be fine.  Maybe the error message should start you need to create your realm before you can sync it's membership.
```

But it might not be accurate in some situations. E.g. if you set up your realm programmatically via `do_create_realm` for example, you end up having a realm - just no users for ldap sync to operate on.

Reply:
Even the `do_create_realm` management command still creates a user.

Reply:
@timabbott Hmm, just to clarify here -  technically it does, but only the realm **bots** unless I'm missing something? And bots are excluded from ldap sync, so I was just thinking of non-bot users.